core/src/main/java/com/threerings/getdown/data/Application.java

```diff
@@ -961,12 +962,20 @@ public Process createProcess (boolean optimum)
         }
 
         // pass along our proxy settings
-        String proxyHost;
-        if ((proxyHost = System.getProperty("http.proxyHost")) != null) {
-            args.add("-Dhttp.proxyHost=" + proxyHost);
-            args.add("-Dhttp.proxyPort=" + System.getProperty("http.proxyPort"));
-            args.add("-Dhttps.proxyHost=" + proxyHost);
-            args.add("-Dhttps.proxyPort=" + System.getProperty("http.proxyPort"));
+        if (proxy != Proxy.NO_PROXY && proxy.address() instanceof InetSocketAddress) {
```

Comment:
if proxy is of type SOCKS then this is illegal to pass it as http proxy, it should be passed as ``-DsocksProxyHost``. see https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html. so condition is wrong. it also would be better to initialize ``proxyHost`` and new ``proxyPort`` vars once and avoid repeating args.add 2 times

Reply:
Because **Application.proxy** is only initialized through 

**ProxyUtil.initProxy** with **Proxy.Type.HTTP**, this is sufficient for the moment.

Reply:
Or did you simply mean a Proxy.Type.HTTP test?

Reply:
yes, at a minimum, i didn't mean to support socks proxy case. but still the last word is up to @samskivert :smile: 

PS: but anyway ProxyUtil resides in upper-level launcher module which depends on core so it would be incorrect to assume anything regarding acceptable values in app.proxy. one can depend on core w/o launcher (as we do for instance).